packages/push/examples/smallest/main.rs

```diff
@@ -0,0 +1,215 @@
+pub mod args;
+
+use std::ops::Not;
+
+use anyhow::{ensure, Result};
+use clap::Parser;
+use ec_core::{
+    distributions::{collection::ConvertToCollectionGenerator, conversion::IntoDistribution},
+    generation::Generation,
+    individual::{ec::WithScorer, scorer::FnScorer},
+    operator::{
+        genome_extractor::GenomeExtractor,
+        genome_scorer::GenomeScorer,
+        mutator::Mutate,
+        selector::{best::Best, lexicase::Lexicase, Select, Selector},
+        Composable,
+    },
+    test_results::{self, TestResults},
+};
+use ec_linear::mutator::umad::Umad;
+use push::{
+    evaluation::cases::{Case, Cases, WithTargetFn},
+    genome::plushy::{GeneGenerator, Plushy},
+    instruction::{
+        variable_name::VariableName, BoolInstruction, ExecInstruction, IntInstruction,
+        PushInstruction,
+    },
+    push_vm::{program::PushProgram, push_state::PushState, stack::StackError, HasStack, State},
+};
+use rand::{distributions::Distribution, rngs::ThreadRng, thread_rng, RngCore};
+use strum::IntoEnumIterator;
+
+use crate::args::{Args, RunModel};
+
+// An input for this problem is a tuple of four `i64`s.
+#[derive(Copy, Clone)]
+struct Input(i64, i64, i64, i64);
```

Comment:
do we want to use 
```suggestion
struct Input([i64;4]);
```

here instead?

Reply:
And if we do that, it might just make sense to do `type Input = [i64;4];` instead